cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller_test.go

```diff
@@ -112,3 +113,45 @@ func TestFindMachineByID(t *testing.T) {
 		t.Fatalf("expected findMachine() to return nil")
 	}
 }
+
+func TestFindNodeByNodeName(t *testing.T) {
+	controller := mustCreateTestController(t)
+
+	testNode := &apiv1.Node{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "ip-10-0-18-236.us-east-2.compute.internal",
+		},
+	}
+
+	controller.nodeInformer.GetStore().Add(testNode)
+
+	// Verify inserted node can be found
+	node, err := controller.findNodeByNodeName("ip-10-0-18-236.us-east-2.compute.internal")
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	if node == nil {
+		t.Fatal("expected a node")
+	}
+
+	// Verify node is identical to that added to the store
+	if !reflect.DeepEqual(*node, *testNode) {
+		t.Fatalf("expected %+v, got %+v", testNode, node)
+	}
+
+	// Verify that a successful findNodeByNodeName returns a DeepCopy().
+	if node == testNode {
```

Comment:
Object address never matters. If node is deep copy of testNode, it will have different address for sure. So this check should fail actually.

Reply:
And it if it doesn't because in the future somebody removes `.DeepCopy()` then this test would fail. the function semantics would fail. this is what I'm trying to assert.